### PR TITLE
JSON-RPC basic authentication

### PR DESCRIPTION
**Pull request by riptl**
Solves #8 

Not tested yet, I'll do that later.

### CODE REVIEW

File: js/settings.js [hunk 1]
```diff
@@ -15,7 +17,11 @@
         switch (service_slug) {
             case 'nimiq_watch': break;
             case 'json_rpc':
-                $('#woocommerce_nimiq_gateway_jsonrpc_url').closest('tr').removeClass('hidden');
+                [
+                    '#woocommerce_nimiq_gateway_jsonrpc_url',
+                    '#woocommerce_nimiq_gateway_jsonrpc_username',
+                    '#woocommerce_nimiq_gateway_jsonrpc_password'
+                ].forEach(tag => $(tag).closest('tr').removeClass('hidden'));
```
Comment (sisou) [inline js/settings.js]: You can but the list of IDs inside the jquery constructor. Jquery can work with a set of elements. That would prevent having to create a new jquery instance for each field.

File: validation_services/json_rpc.php [hunk 2]
```diff
@@ -53,7 +53,20 @@ public function load_transaction( $transaction_hash ) {
 
         $call = '{"jsonrpc":"2.0","method":"getTransactionByHash","params":["' . $transaction_hash . '"],"id":42}';
 
-        $api_response = wp_remote_post( $this->api_domain, array( 'body' => $call ) );
+        $username = $gateway->get_option( 'jsonrpc_username' );
+        $password = $gateway->get_option( 'jsonrpc_password' );
+        $headers = array( );
+    
+        if ( !empty( $username ) || !empty( $password ) ) {
+            $authorization = 'Basic ' . base64_encode( $username . $password );
```
Comment (sisou) [inline validation_services/json_rpc.php]: Missing a colon `:` between the username and password here.

**Review by sisou — CHANGES_REQUESTED** [review]

File: validation_services/json_rpc.php (hunk 2 above)
Comment (riptl) [reply validation_services/json_rpc.php]: thanks!

Comment (sisou) [reply validation_services/json_rpc.php]: I meant, between the *username* and *password* ;p

Comment (sisou) [reply validation_services/json_rpc.php]: Compare again with https://github.com/nimiq-network/core/blob/master/clients/nodejs/modules/JsonRpcServer.js#L165

Comment (riptl) [reply validation_services/json_rpc.php]: @sisou That's so embarrassing 😂 I know how Basic auth works, I swear

Comment (riptl) [reply validation_services/json_rpc.php]: I'm amending the last commit so no-one sees it

**Comment by sisou** [comment]
Have you tested this yet in a local setup, @terorie?

**Review by sisou — APPROVED** [review]

**Review by sisou — APPROVED** [review]

**Pull request merged**